server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -102,10 +100,6 @@ static TermVectorsResponse getTermVectors(IndexShard indexShard, TermVectorsRequ
             /* from an artificial document */
             if (request.doc() != null) {
                 termVectorsByField = generateTermVectorsFromDoc(indexShard, request);
-                // if no document indexed in shard, take the queried document itself for stats
-                if (topLevelFields == null) {
-                    topLevelFields = termVectorsByField;
-                }
```

Comment:
I cannot find a relation to the dfs removal for this removed block, probably unrelated, but would you briefly explain to me why this can be removed?

Reply:
`topLevelFields` is never null, so this is a no-op, according to IntelliJ.